### PR TITLE
added some initializations variables as well as free and nullify

### PR DESCRIPTION
**Pull request by BitExplorer**
Just throwing out a PR to address to ensure the free call does not have any issues.

### CODE REVIEW

**Comment by lfos** [comment]
Could you please explain why this is needed or a good idea?

File: jpeg-compare.c [hunk 1]
```diff
@@ -96,13 +97,15 @@ int compareFastFromBuffer(unsigned char *imageBuf1, long bufSize1, unsigned char
 
     // Cleanup
     free(hash1);
+    hash1 = NULL;
```
Comment (zvezdochiot) [inline jpeg-compare.c]: Maybe?:
```c
*hash1 = NULL;
*hash2 = NULL;
```

Comment (lfos) [reply jpeg-compare.c]: You cannot modify memory after it has been free()'d.

Comment (zvezdochiot) [reply jpeg-compare.c]: Sorry. Blunder.

**Comment by lfos** [comment]
Closing for now. Feel free to reopen with more details.

**Pull request closed without merging**